game/modes.py

```diff
@@ -0,0 +1,143 @@
+from .board import board_matrix as initial_board, get_options, show, update_board
+from .status import check_status
+from .player import get_current_player
+import logging
+import random
+import copy
+
+
+logger = logging.getLogger(__name__)
+
+
+def _find_winning_cell(board_matrix, available_options, sign):
+    '''
+    Creates a copy of the current state of the board and tries all the available options with the sign argument
+    and checks if any choice wins, returns True or False if there is a winning choice, along with the the choice itself.
+    '''
+    is_won = False
+    choice = None
+    for choice in available_options:
+        board_copy = copy.deepcopy(board_matrix)
+        new_board = update_board(board_copy, choice, sign)
+        is_won, _ = check_status(new_board)
+        if is_won:
+            break
+    return is_won, choice
+
+
+def _check_cell(cell_number, board):
+    '''
+    returns the Data ('x', 'o' or None) from the given cell in the given board
+    '''
+    row = int((cell_number-.5) // 3)
+    col = (cell_number - row * 3) - 1
+    return board[row][col]
+
+
+def _computer_move(available_options, board_matrix, last_move, difficulty):
+    '''
+    difficulties: 'easy' picks randomly from the available options
+    'medium' checks to for a winning move for itself, or blocks player from winning on next move. 
+    if there is no winning move next move, it chooses a random available option
+    'hard' checks for win or blocks player like the medium difficulty, but also strategically picks a move
+    if there is no winning or blocking move
+    
+    return:: choice
+    '''
```

Comment:
1. naming convention (unless there is another reason for using underscore at the beginning of the function's name)
2. PEP warning for triple double-quotes

```suggestion
def _computer_move(available_options, board_matrix, last_move, difficulty):
    """
    difficulties: 'easy' picks randomly from the available options
    'medium' checks to for a winning move for itself, or blocks player from winning on next move. 
    if there is no winning move next move, it chooses a random available option
    'hard' checks for win or blocks player like the medium difficulty, but also strategically picks a move
    if there is no winning or blocking move
    
    return:: choice
    """
```